dashboard/src/components/AppView/AppView.tsx

```diff
@@ -215,34 +215,38 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
       secretRefs: [],
     };
     resources.forEach(i => {
-      const item = i as IResource;
-      const resource = { isFetching: true, item };
-      switch (i.kind) {
-        case "Deployment":
-          result.deployRefs.push(new ResourceRef(resource.item, releaseNamespace));
-          break;
-        case "Service":
-          result.serviceRefs.push(new ResourceRef(resource.item, releaseNamespace));
-          break;
-        case "Ingress":
-          result.ingressRefs.push(new ResourceRef(resource.item, releaseNamespace));
-          break;
-        case "Secret":
-          result.secretRefs.push(new ResourceRef(resource.item, releaseNamespace));
-          break;
-        case "List":
-          // A List can contain an arbitrary set of resources so we treat them as an
-          // additional manifest. We merge the current result with the resources of
-          // the List, concatenating items from both.
-          _.assignWith(
-            result,
-            this.parseResources((i as IK8sList<IResource, {}>).items, releaseNamespace),
-            // Merge the list with the current result
-            (prev, newArray) => prev.concat(newArray),
-          );
-          break;
-        default:
-          result.otherResources.push(item);
+      // The item may be a list
+      const itemList = i as IK8sList<IResource, {}>;
+      if (itemList.items) {
```

Comment:
I wonder if, to be safe, we should also check if `kind ~= /List/`?

Reply:
it seems more probably to me to find components that have an array of items and doesn't have `List` in the name (e.g. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#downwardapivolumesource-v1-core)

Reply:
The DownwardAPIVolumeSource isn't a top-level API resource, it's a type used in other resources. I'm fairly sure all Kubernetes core API resources use `List` in the name for list types. It is possible that a CRD can define a different name for list kinds (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#customresourcedefinitionnames-v1beta1-apiextensions-k8s-io), though note that the default is `<kind>List`.